downstairs/src/region.rs

```diff
@@ -114,6 +114,7 @@ pub struct Region {
     log: Logger,
 
     /// Thread pool for doing long-running CPU work outside the Tokio runtime
+    #[allow(unused)]
     pool: rayon::ThreadPool,
```

Comment:
maybe this could be `#[cfg(not(feature = "omicron-build"))]` instead of `allow(unused)`?

Reply:
Removing it entirely is awkward, because then I have to sprinkle `#[cfg(not(feature = "omicron-build"))]` in a bunch of other places.  How about making the `allow(unused)` conditional instead?
```rust
#[cfg_attr(feature = "omicron-build", allow(unused))]
```